### PR TITLE
Avoid some potential test race conditions by being explicit about which tests need DismissAutoManageDialog()…

### PR DESCRIPTION
**Pull request by bspratt**
… and which do not.

### CODE REVIEW

File: pwiz_tools/Skyline/TestUtil/TestFunctional.cs [hunk 1]
```diff
@@ -2204,32 +2204,25 @@ public static void PasteSmallMoleculeListNoAutoManage()
...
-            if (expectAutoManage ?? false)
+            var autoManageDlg = WaitForOpenForm<MultiButtonMsgDlg>();
+            if (autoManageDlg == null)
             {
-                wantAutoManageDlg = WaitForOpenForm<MultiButtonMsgDlg>();
+                AssertEx.Fail($@"Expected dialog with message like ""${Resources.SkylineWindow_ImportMassList_Do_you_want_to_use_the_document_settings_to_automanage_these_new_transitions}""");
             }
-            else
+            // Make sure we haven't intercepted some other dialog
+            else if (IsFormattedMessage(Resources.SkylineWindow_ImportMassList_Do_you_want_to_use_the_document_settings_to_automanage_these_new_transitions, autoManageDlg.Message))
```
Comment (brendanx67) [inline pwiz_tools/Skyline/TestUtil/TestFunctional.cs:2215]: This feels like it should be replaced with a simple assert, getting rid of the if clause entirely and the need to call AssertEx.Fail() yourself.

**Review by brendanx67 — CHANGES_REQUESTED** [review]
I think the use of conditions versus straight assertions could be improved in DismissAutoManageDialog()

**Review by brendanx67 — APPROVED** [review]

**Comment by brendanx67** [comment]
I simplified the DismissAutoManageDialog() function a bit more and removed IsFormattedMessage() in favor of using AssertEx.AreComparableStrings(). I also ended up improving diagnostics to help understand failures I saw running parallel tests. Sorry for adding unrelated changes to your PR.

**Comment by bspratt** [comment]
I like my implementation better as it tells you exactly what the test was waiting for (the AutoManage dialog) rather than just "I was expecting a generic message dialog", but not enough to make a fuss about it.

**Comment by brendanx67** [comment]
Yeah. You can learn that from the call stack pretty easily. Not worth adding code complexity to put that in the message.

**Comment by brendanx67** [comment]
I set the timeout to 1 millisecond to see what a failure report looks like. Funny that by the time it starts reporting the failure the MultiButtonMsgDlg is present, but you can see that DismissAutoManageDialog is in the callstack.

Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Assert.Fail failed. Timeout 0 seconds exceeded in WaitForOpenForm(MultiButtonMsgDlg). Open forms: SequenceTreeForm (), SkylineWindow (Skyline), MultiButtonMsgDlg ()
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.HandleFailure(String assertionName, String message)
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.Fail(String message, Object[] parameters)
   at pwiz.SkylineTestUtil.AbstractFunctionalTest.WaitForOpenForm[TDlg](Int32 millis) in C:\proj\pwiz\pwiz_tools\Skyline\TestUtil\TestFunctional.cs:line 783
   at pwiz.SkylineTestUtil.AbstractFunctionalTest.DismissAutoManageDialog() in C:\proj\pwiz\pwiz_tools\Skyline\TestUtil\TestFunctional.cs:line 2198

**Pull request merged**